sdk/communication/azure-communication-administration/azure/communication/administration/_phone_number_administration_client.py

```diff
@@ -387,40 +420,75 @@ def list_all_releases(
         )
 
     @distributed_trace
-    def get_search_by_id(
+    def get_reservation_by_id(
         self,
-        search_id,  # type: str
+        reservation_id,  # type: str
         **kwargs  # type: Any
     ):
-        # type: (...) -> PhoneNumberSearch
-        """Get search by search id.
+        # type: (...) -> PhoneNumberReservation
+        """Get reservation by reservation id.
 
-        :param search_id: The search id to be searched for.
-        :type search_id: str
-        :rtype: ~azure.communication.administration.PhoneNumberSearch
+        :param reservation_id: The reservation id to get reservation.
+        :type reservation_id: str
+        :rtype: ~azure.communication.administration.PhoneNumberReservation
         """
         return self._phone_number_administration_client.phone_number_administration.get_search_by_id(
-            search_id,
+            search_id=reservation_id,
             **kwargs
         )
 
     @distributed_trace
-    def create_search(
+    def begin_reserve_phone_numbers(
         self,
         **kwargs  # type: Any
     ):
-        # type: (...) -> CreateSearchResponse
-        """Creates a phone number search.
-
-        :keyword azure.communication.administration.CreateSearchOptions body:
-        An optional parameter for defining the search options.
-        The default is None.
-        :rtype: ~azure.communication.administration.CreateSearchResponse
+        # type: (...) -> LROPoller
+        """Begins creating a phone number search to reserve phone numbers.
+        Caller must provide either options, or continuation_token keywords to use the method.
+        If both options and continuation_token are specified, only continuation_token will be used to
+        restart a poller from a saved state, and keyword options will be ignored.
+        :keyword azure.communication.administration.CreateSearchOptions options: reservation options.
```

Comment:
Do you think `create_search_options` might be a better name for this kwarg?

Reply:
In fact, can we flatten CreateSearchOptions?

Reply:
@xiangyan99 
I have synced with other team members regarding this question for flattening [CreateSearchOptions](https://github.com/Azure/azure-sdk-for-python/blob/4096074feddbb76469a6816712a87fd4c6c5c1ad/sdk/communication/azure-communication-administration/azure/communication/administration/_phonenumber/_generated/models/_models_py3.py#L162).

There are 6 params in CreateSearchOptions: 
  display_name, description, phone_plan_ids, area_code, quantity and location_options.

5 of these 6 params are mandatory all the time: display_name, description, phone_plan_ids, area_code, quantity
The only "optional" param is location_options: location_options is required when the provided phone_plan_ids are local phone plans. When the provided phone_plan_ids are toll-free, the location_options is not needed. 

Any thought on this situation? Thanks.

Reply:
I think we want to have required arguments as positional arguments and put the optional one in kwargs. By doing that, user can have better doc/intellisense experience.

Reply:
@turalf  You mean `options` -> `begin_reservation_options`? Since the `options` is a keyword of `begin_reserve_phone_numbers()`, I feel the belonging of the keyword `options` to `begin_reserve_phone_numbers()` already represents the meaning of the options for phone number reservation. It is redundant to have the prefix `begin_reservation_` to `options`. Just some thoughts here. What do you think?

Reply:
@xiangyan99 Thanks. Make a lot of sense. But some people may have concern that we have too many positional arguments (5 args). Is 5 positional arguments OK in Python? I guess it may vary in the language.

Reply:
In fact 5 is on the fence. @johanste How do you think?

Reply:
@xiangyan99 Johan is out of office until 11/16. Maybe we can reach to another person?
 The [Python Guidance](https://azure.github.io/azure-sdk/python_introduction.html) says "have methods that require more than five positional parameters."

Reply:
I have created a new issue and will make a separate PR for the discussion & following-up changes. 
https://github.com/Azure/azure-sdk-for-python/issues/14959